packages/core/test-project/android/app/src/test/java/io/embrace/rnembracecoretest/RNEmbraceCoreTest.kt

```diff
@@ -4,36 +4,157 @@ import android.os.SystemClock
+                whenever(mockEmbrace.generateW3cTraceparent()).thenReturn(mockTraceparent)
+
+                val url = "http://request.com/v1/all"
+                val httpMethod = "GET"
+                val startTime = currentTimeMillis().toDouble()
+                val endTime = currentTimeMillis().toDouble() + 400000
+                val bytesSent = 12938
+                val bytesReceived = 199
+                val statusCode = 200
+
+                embraceModule.logNetworkRequest(
+                    url,
+                    httpMethod,
+                    startTime,
+                    endTime,
+                    bytesSent,
+                    bytesReceived,
+                    statusCode,
+                    promise,
+                )
+
+                verify(mockEmbrace).generateW3cTraceparent()
 
-        // embrace starts without issues
-        verify(promise, times(1)).resolve(true)
+                mockedStatic.verify {
+                    EmbraceNetworkRequest.fromCompletedRequest(
+                        url,
+                        HttpMethod.GET,
+                        startTime.toLong(),
+                        endTime.toLong(),
+                        bytesSent.toLong(),
+                        bytesReceived.toLong(),
+                        statusCode,
+                        null,
+                        mockTraceparent,
+                        null
+                    )
+                }
+
+                verify(promise, times(1)).resolve(true)
+            }
+        }
+    }
+
+    @Test
+    fun testLogNetworkClientError() {
+        val context: ReactApplicationContext = mock()
+        val embraceModule = EmbraceManagerModule(context)
+
+        Mockito.mockStatic(Embrace::class.java).use { mockedEmbraceStatic ->
+            mockedEmbraceStatic.`when`<Any>(Embrace::getInstance).thenReturn(mockEmbrace)
+
+            Mockito.mockStatic(EmbraceNetworkRequest::class.java).use { mockedStatic: MockedStatic<EmbraceNetworkRequest> ->
+                val mockTraceparent = "traceparent-123"
+                whenever(mockEmbrace.generateW3cTraceparent()).thenReturn(mockTraceparent)
+
+                val url = "http://request.com/v1/all"
+                val httpMethod = "POST"
+                val startTime = currentTimeMillis().toDouble()
+                val endTime = currentTimeMillis().toDouble() + 400000
+                val errorType = "Bad Request"
+                val errorMessage = "Error message"
+
+                embraceModule.logNetworkClientError(
+                    url,
+                    httpMethod,
+                    startTime,
+                    endTime,
+                    errorType,
+                    errorMessage,
+                    promise,
+                )
+
+                verify(mockEmbrace).generateW3cTraceparent()
+
+                mockedStatic.verify {
+                    EmbraceNetworkRequest.fromIncompleteRequest(
```

Comment:
in the interest of mocking less I think we should setup a log exporter on an actual embrace SDK instance similar to how the otlp package does it and then verify that the traceparent is set based on the attributes of the log that was generated by `logNetworkClientError`